core/components/atoms/_chip/index.tsx

```diff
@@ -82,49 +94,67 @@ export const GenericChip = (props: GenericChipProps) => {
           <Text data-test="DesignSystem-GenericChip--Text" color={textColor} className="Chip-text">
             {label}
           </Text>
-        </>
+        </div>
       );
     }
     return label;
   };
 
   return (
-    // TODO(a11y)
-    // eslint-disable-next-line
-    <div
-      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-      tabIndex={disabled ? -1 : 0}
-      data-test="DesignSystem-GenericChip--GenericChipWrapper"
-      {...baseProps}
-      className={`Chip-wrapper ${className}`}
-      onClick={onClickHandler}
-    >
-      {icon && (
-        <Icon
-          data-test="DesignSystem-GenericChip--Icon"
-          name={icon}
-          type={iconType}
-          appearance={iconAppearance('left')}
-          className={iconClass('left')}
-        />
-      )}
-      {renderLabel()}
-      {clearButton && (
-        <div
-          role="button"
-          onClick={onCloseHandler}
-          tabIndex={disabled ? -1 : 0}
-          onKeyDown={onKeyDownHandler}
-          className={iconClass('right')}
-          data-test="DesignSystem-GenericChip--clearButton"
-        >
-          <Icon name="clear" appearance={iconAppearance('right')} className="p-2" />
-        </div>
-      )}
+    <div>
+      <Tooltip
+        showTooltip={isTextTruncated}
+        data-test="DesignSystem-GenericChip--Tooltip"
+        tooltip={
+          labelPrefix
+            ? `${labelPrefix} ${typeof label === 'string' ? label : ''}`
+            : typeof label === 'string'
+            ? label
```

Comment:
`typeof label === 'string' ? label : ''` this same conditional statement is used twice please refactor this for cleaner code.

Reply:
done